src/tinyvec.rs

```diff
@@ -1411,3 +1450,35 @@ where
     write!(f, "]")
   }
 }
+
+#[cfg(feature = "serde")]
+struct TinyVecVisitor<A: Array>(PhantomData<A>);
+
+#[cfg(feature = "serde")]
+impl<'de, A: Array> Visitor<'de> for TinyVecVisitor<A>
+where
+  A: Default,
+  A::Item: Deserialize<'de>,
+{
+  type Value = TinyVec<A>;
+
+  fn expecting(
+    &self,
+    formatter: &mut core::fmt::Formatter,
+  ) -> core::fmt::Result {
+    formatter.write_str("a sequence")
+  }
+
+  fn visit_seq<S>(self, mut seq: S) -> Result<Self::Value, S::Error>
+  where
+    S: SeqAccess<'de>,
+  {
+    let mut new_arrayvec: ArrayVec<A> = ArrayVec::new();
+
+    while let Some(value) = seq.next_element()? {
+      new_arrayvec.push(value);
```

Comment:
This will panic if deserializing something that doesn't fit into the `Arrayvec`. Should probably use `seq.size_hint()` to pre-alloc a `TinyVec` to push into, defaulting to the array if unknown, which will then auto-expand onto the heap if necessary.

Reply:
... I don't know how I missed the fact that `TinyVec::with_capacity` exists.